### PR TITLE
return backfill to order by date, batch size 1000

### PR DESCRIPTION
**Pull request by drewgillies**
Backfill job was ordered in such a way as to not work with cursoring. There was a filtering condition removed which also needed to be reinstated. I also upped the batching size to 1000 for the TFM backfill.

### CODE REVIEW

File: src/GalleryTools/Commands/BackfillCommand.cs [hunk 1]
```diff
@@ -122,10 +122,11 @@ public async Task Collect(SqlConnection connection, Uri serviceDiscoveryUri, Dat
                 {
                     packages = packages.Include(QueryIncludes);
                 }
-                
+
                 packages = packages
                     .Where(p => p.Created < lastCreateTime && p.Created > startTime)
-                    .OrderBy(p => p.PackageRegistration.Id);
+                    .Where(p => p.PackageStatusKey == PackageStatus.Available || p.PackageStatusKey == PackageStatus.Validating)
```
Comment (joelverhagen) [inline src/GalleryTools/Commands/BackfillCommand.cs:128]: `Validating` packages won't be available in v3-flatcontainer. Could we stick to `PackageStatus.Available`?

**Review by joelverhagen — CHANGES_REQUESTED** [review]
Unsure about the validating packages addition.

File: src/GalleryTools/Commands/BackfillCommand.cs (hunk 1 above)
Comment (drewgillies) [reply src/GalleryTools/Commands/BackfillCommand.cs:128]: @joelverhagen Great insight. Thank you. Note that this is for all backfill jobs--this is the parent class.

**Comment by drewgillies** [comment]
>Unsure about the validating packages addition.
@joelverhagen note that this is removed code that I reintroduced. It could be an implementation detail (i.e. a flag) set in the subclasses if we like, defaulting to false.

File: src/GalleryTools/Commands/BackfillTfmMetadataCommand.cs [hunk 2]
```diff
@@ -28,10 +31,34 @@ public static void Configure(CommandLineApplication config)
...
-            var supportedFrameworks = _packageService.GetSupportedFrameworks(nuspecReader, files);
+
+            // We wrap this int a try catch because fetching supported frameworks on existing packages can sometimes throw due to format errors
+            // (usually portable TFM formatting). In this case we'll clear the TFMs.
+            var supportedFrameworks = Enumerable.Empty<NuGetFramework>();
+            try
+            {
+                supportedFrameworks = _packageService.GetSupportedFrameworks(nuspecReader, files);
+            }
+            catch (Exception e)
```
Comment (joelverhagen) [inline src/GalleryTools/Commands/BackfillTfmMetadataCommand.cs:42]: We can catch a more specific exception here, right? I think it's a specific one that is caused by the bad portables.

Comment (joelverhagen) [reply src/GalleryTools/Commands/BackfillTfmMetadataCommand.cs:42]:
```suggestion
            catch
```
if we end up using a generic catch-all like the other places

**Review by joelverhagen — APPROVED** [review]
Try to catch a more specific exception if possible. But otherwise looks good.

File: src/GalleryTools/Commands/BackfillTfmMetadataCommand.cs (hunk 2 above)
Comment (drewgillies) [reply src/GalleryTools/Commands/BackfillTfmMetadataCommand.cs:42]: It's a little generic--it's an `ArgumentException` which is why I opted for catching all. But I'm happy to make it explicit as exceptions not of this type will fall through and be logged, which could be useful.

**Pull request merged**